jupyterhub/tests/selenium/test_browser.py

```diff
@@ -209,36 +225,686 @@ async def test_open_url_login(
+
+    # select the token duration
+    select_element = browser.find_element(*TokenPageLocators.LIST_EXP_TOKEN_FIELD)
+    while token_opt != "Never":
+        select = Select(select_element)
+        select.select_by_visible_text(token_opt)
+        break
+    if note == True:
+        send_text(browser, TokenPageLocators.INPUT_TOKEN, note_value)
+        assert (
+            browser.find_element(*TokenPageLocators.INPUT_TOKEN).get_attribute('value')
+            == note_value
+        )
+    else:
+        pass
+    await asyncio.sleep(1)
+    click(browser, TokenPageLocators.BUTTON_API_REQ)
+    # refresh the page
+    await in_thread(browser.get, browser.current_url)
+    # API Tokens table: verify that elements are displayed
+    assert is_displayed(browser, TokenPageLocators.TABLE_API)
+    assert is_displayed(browser, TokenPageLocators.TABLE_API_HEAD)
+    assert is_displayed(browser, TokenPageLocators.TABLE_API_BODY)
+    assert int(len(elements_row_head(browser))) == 1
+    assert int(len(elements_row_body(browser))) == 1
+    List_rows_head = []
+    List_col_head = []
+    # verify head of table rows
+    for row_head in elements_row_head(browser):
+        i = row_head.text
+        List_rows_head.append(i)
+        columns_head = elements_row_head(browser)[0].find_elements(
+            *TokenPageLocators.TABLE_API_COLUMNS
+        )
+        for column_head in columns_head:
+            j = column_head.text
+            List_col_head.append(j)
+
+    assert List_col_head == TokenPageLocators.TABLE_API_HEAD_LIST
+    # verify body of table
+    Dict_body_rows = {}
+    Dict_body_columns = {}
+    for i in range(len(elements_row_body(browser))):
+        await asyncio.sleep(1)
+        Dict_body_rows[i] = elements_row_body(browser)[i].text
+        columns_body = elements_row_body(browser)[i].find_elements(
+            *TokenPageLocators.TABLE_API_COLUMNS
+        )
+        for j in range(len(columns_body)):
+            Dict_body_columns[j] = columns_body[j].text
+    assert is_displayed(browser, TokenPageLocators.TABLE_API_COLUMNS)
+    assert int(len(columns_head)) == 4
+    assert int(len(columns_body)) == 5
+
+    if note == False:
+        assert Dict_body_columns.get(0) == TokenPageLocators.TEXT_WOUT_NOTE
+    else:
+        assert Dict_body_columns.get(0) == note_value
+    if token_opt == "Never":
+        assert Dict_body_columns.get(3) == "Never"
+    elif token_opt == "1 Hour":
+        assert Dict_body_columns.get(3) == "in an hour"
+    elif token_opt == "1 Day":
+        assert Dict_body_columns.get(3) == "in a day"
+    elif token_opt == "1 Week":
+        assert Dict_body_columns.get(3) == "in 7 days"
+    # verify that the button for revoke is presented
+    buttons = elements_table_body(browser).find_elements(
+        *TokenPageLocators.BUTTON_REVOKE
+    )
+    for button in buttons:
+        button = button.text
+    assert button == columns_body[len(columns_body) - 1].text
+    assert int(len(buttons)) == 1
```

Comment:
Let's test at the end with a db query that the tokens were actually created with the right expiration.